sentence_transformers/sampler.py

```diff
@@ -53,32 +72,31 @@ def __init__(
         if self.batch_size % 2 == 1:
             raise ValueError("The batch size for `GroupByLabelBatchSampler` must be divisible by 2.")
 
-        for column_name in valid_label_columns or []:
-            if column_name in dataset.column_names:
-                labels = dataset["label"]
-                break
-        else:
-            raise ValueError(f"None of the valid_label_columns {valid_label_columns} are in the dataset.")
-
-        del dataset
+        labels = self._determine_labels_to_use(dataset, valid_label_columns)
         groups = defaultdict(list)
         for sample_idx, label in enumerate(labels):
             groups[label].append(sample_idx)
 
         self.groups = {
-            label: sample_indices[:num_samples]
-            for label, sample_indices in groups.items()
-            if (num_samples := len(sample_indices) // 2)
+            label: sample_indices[: len(sample_indices) // 2 * 2] for label, sample_indices in groups.items()
```

Comment:
I believe this optimization isn't ideal, as `sample_indices[: len(sample_indices) // 2 * 2]` can now be an empty list. Though it doesn't matter much (it only costs an unnecessary iteration in the `__iter__`), it might be best to keep the `if`. I put that filter in there to avoid this edge case

Reply:
Valid point! I changed it back to the way you proposed.